c_emulator/riscv_callbacks_if.h

```diff
@@ -8,31 +8,39 @@
 class callbacks_if {
 public:
   virtual ~callbacks_if() = default;
-
   // TODO: finding ways to improve the format
-  virtual void fetch_callback(sbits opcode) = 0;
-  virtual void mem_write_callback(const char *type, sbits paddr, uint64_t width,
-                                  lbits value)
-      = 0;
-  virtual void mem_read_callback(const char *type, sbits paddr, uint64_t width,
-                                 lbits value)
-      = 0;
-  virtual void mem_exception_callback(sbits paddr, uint64_t num_of_exception)
-      = 0;
-  virtual void xreg_full_write_callback(const_sail_string abi_name, sbits reg,
-                                        sbits value)
-      = 0;
-  virtual void freg_write_callback(unsigned reg, sbits value) = 0;
-  virtual void csr_full_write_callback(const_sail_string csr_name, unsigned reg,
-                                       sbits value)
-      = 0;
-  virtual void csr_full_read_callback(const_sail_string csr_name, unsigned reg,
-                                      sbits value)
-      = 0;
-  virtual void vreg_write_callback(unsigned reg, lbits value) = 0;
-  virtual void pc_write_callback(sbits new_pc) = 0;
-  virtual void redirect_callback(sbits new_pc) = 0;
-  virtual void trap_callback(bool is_interrupt, fbits cause) = 0;
+  virtual void fetch_callback([[maybe_unused]] sbits opcode) { };
```

Comment:
Lots of extra semicolons.

```suggestion
  virtual void fetch_callback([[maybe_unused]] sbits opcode) { }
```